aviary/subsystems/propulsion/engine_deck.py

```diff
@@ -1331,6 +1355,235 @@ def _count_data(self):
+        engine_group.add_subsystem('interpolation',
+                                   engine,
+                                   promotes_inputs=['*'],
+                                   promotes_outputs=['*'])
+
+        if power_type is SHAFT_POWER_CORRECTED:
+            self._uncorrect_shaft_power(engine_group, num_nodes=num_nodes)
+
+        scaling_group = om.Group()
+        variables_to_scale = ['shaft_power', 'tailpipe_thrust',
+                              'nox_rate', 'electric_power', 'thrust_net_max']
+        for variable in variables_to_scale:
+            self.add_scaling_exec_comp(scaling_group, variable, num_nodes=num_nodes)
+        self.add_scaling_exec_comp(scaling_group, 'fuel_flow_rate', num_nodes=num_nodes,
+                                   alias=Dynamic.Mission.FUEL_FLOW_RATE_NEGATIVE)
+        self.add_scaling_exec_comp(scaling_group, 'thrust_net',
+                                   num_nodes=num_nodes, alias='unused')
+
+        engine_group.add_subsystem('engine_scaling',
+                                   subsys=scaling_group,
+                                   promotes_inputs=['*'],
+                                   promotes_outputs=['*'])
+
+        if self.prop_model:
+            engine_group.add_subsystem(
+                'propeller_model',
+                self.prop_model,
+                promotes_inputs=[Dynamic.Mission.SHAFT_POWER],
+                promotes_outputs=['prop_thrust'],
+            )
+        else:
+            self._add_dummy_prop(engine_group, num_nodes)
+
+        engine_group.add_subsystem(
+            'total_thrust',
+            om.ExecComp(
+                'total_thrust = prop_thrust + tailpipe_thrust',
+                total_thrust={'units': 'lbf', 'shape': num_nodes},
+                prop_thrust={'units': 'lbf', 'shape': num_nodes},
+                tailpipe_thrust={'val': np.zeros(num_nodes), 'units': 'lbf'},
+            ),
+            promotes_inputs=['prop_thrust', 'tailpipe_thrust'],
+            promotes_outputs=[('total_thrust', Dynamic.Mission.THRUST)],
+        )
+
+        return engine_group
+
+    def add_scaling_exec_comp(self, grp: om.Group, variable: str, units=None, num_nodes=1, alias=None):
+        if alias is None:
+            alias = variable
+        if units is None:
+            _, units = self.options.get_item(variable)
+        grp.add_subsystem(
+            variable+'_scaling',
+            om.ExecComp(
+                'scaled_variable = variable_unscaled * scale_factor',
+                scaled_variable={'shape': num_nodes, 'units': units},
+                variable_unscaled={'shape': num_nodes, 'units': units},
+                scale_factor={'val': 1, 'units': 'unitless'},
+            ),
+            promotes_inputs=[
+                ('variable_unscaled', variable+'_unscaled'),
+                ('scale_factor', Aircraft.Engine.SCALE_FACTOR)
+            ],
+            promotes_outputs=[('scaled_variable', alias)]
```

Comment:
All of these execcomps should have the argument `has_diag_partials = True`, since each element in num_nodes is decoupled from the others.